formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/internal/ProtobufTaggedEncoder.kt

```diff
@@ -27,7 +27,11 @@ internal abstract class ProtobufTaggedEncoder : ProtobufTaggedBase(), Encoder, C
 
     protected open fun encodeTaggedInline(tag: ProtoDesc, inlineDescriptor: SerialDescriptor): Encoder = this.apply { pushTag(tag) }
 
-    public final override fun encodeNull(): Unit = encodeTaggedNull()
+    public final override fun encodeNull() {
+        if (!nullIsAcceptable) {
+            throw SerializationException("'null' is not supported in ProtoBuf")
```

Comment:
Probably we need to pass additional information here, tag or descriptor, to make an exception easier to understand e.g. 'null is not supported for property X because it has default value/nested collection/etc'